### DISCUSSION TITLE
PostgreSQL connection settings for CockroachDB Cloud

### DISCUSSION THREAD

**Discussion by ptman** [nocodb/nocodb#736] [Q&A]
You're given a connection string like `postgresql://$USER:$PASSWORD@$HOST:$PORT/defaultdb?sslmode=verify-full&sslrootcert=$HOME/.postgresql/root.crt&options=--cluster%3D$NAME-$NUM`

Any idea what to input to be able to connect. I checked https://knexjs.org/#Installation-client

**Comment by o1lab** [comment]
You would have to figure what a successful connection looks like with knexjs and cockroacdb

- Run nocodb without using NC_DB variable (to experiment)
- Create new project by connecting to external database 
- Enter cockroach credentials
- Then you can customise knex connection [params like this.](https://www.youtube.com/watch?v=rS0NRwRCjx0)
- If test database connection goes through - its good.

Curious to hear back on how this goes.

Also see how relevant is this discussion in knexjs : https://github.com/knex/knex/issues/2002

**Comment by backslashmatt** [comment]
Not sure if anyone else is trying to connect to a cockroach db but you can do it under pg. If you change the connection json to add the version:
```
  "dataSource": {
    "client": "pg",
    "version": "20.1",
    "connection": {...}
  }
```

It does connect and seems to work.
